playground/lines.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @title Lines
+ * @category 2D
+ */
+ import {
+  Camera, Color,
+  Vector2,
+  Vector3,
+  WebGLEngine
+} from "oasis-engine";
+import { Line, DashLine, LineCap, LineJoin } from "oasis-engine-toolkit";
+import * as dat from 'dat.gui';
+
+const gui = new dat.GUI({
+	name: 'line'
+});
+const controls = {
+	width: 0.02,
+	cap: LineCap.Round,
+	join: LineJoin.Round,
+}
+const widthControl = gui.add(controls, 'width', 0.01, 0.1);
+const capControl = gui.add(controls, 'cap', { butt: LineCap.Butt, round: LineCap.Round, square: LineCap.Square});
+const joinControl = gui.add(controls, 'join', { bevel: LineJoin.Bevel, round: LineJoin.Round, miter: LineJoin.Miter});
+
+widthControl.onChange(change)
+capControl.onChange(change)
+joinControl.onChange(change)
+
+const lines: Line[] = [];
+const colors = [
+	new Color(91 / 255, 143 / 255, 249 / 255), 
+	new Color(92 / 255, 206 / 255, 161 / 255),
+	new Color(246 / 255, 189 / 255, 22 / 255),
+	new Color(170 / 255, 0 / 255, 97 / 255),
+	new Color(0 / 255, 155 / 255, 119 / 255)
+
+]
+function change() {
+	if(lines.length) {
+		lines.forEach(line => {
+			line.cap = controls.cap;
+			line.join = Number(controls.join);
+			line.width = Number(controls.width);
+		})
+	}
+}
```

Comment:
move  14~47 lines to "function showDebug()"

Reply:
fixed